README.md

```diff
@@ -70,21 +77,65 @@ window.addEventListener('portalactivate', e => {
 });
 ```
 
-## Use cases
+If `https://example.com/` does not need the predecessor-adoption capability, then it can instead use the generic APIs for reacting to prerendering activation or other relevant changes, outlined [in that explainer](https://github.com/jeremyroman/alternate-loading-modes/blob/gh-pages/browsing-context.md#javascript-api).
 
-_See the ["Key Scenarios"](./key-scenarios.md) document for more detail on each of these, including visualizations._
+### Navigation transitions
+
+As a more realistic example, consider a page which wants to prerender a link and provide a seamless navigation transition when it's clicked. It can do this by hiding the portal until the link is clicked,
+and then animating the portal to full screen before activation.
+
+This might be written like so:
 
-- **Pre-rendering**: by loading a page in a hidden portal, it is possible to pre-render the destination page. Then, activating the portal will instantly display the pre-rendered document in the same browser tab.
+```html
+<a href="https://example.com/" class="seamless">Click me!</a>
+
+<script>
+for (const link of document.querySelectorAll('a.seamless')) {
+  const portal = document.createElement('portal');
+  portal.src = link.href;
+  portal.hidden = true;
+  portal.style = 'position: fixed; top: 0; left: 0; width: 100px; height: 100px;';
+  document.body.append(portal);
 
-  This requires some care and cooperation from both sides to deal with the restrictions on portaled content, and the transition between portaled and activated states.
+  link.onclick = async e => {
+    if (portal.state === 'closed') {
+      // The content couldn't be portaled, likely because it didn't opt-in.
+      // Let the normal link click go through.
+      return;
+    }
+
+    e.preventDefault();
+
+    // Show the portal, and animate it to the whole viewport over 300 milliseconds.
+    portal.hidden = false;
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate([{ width: '100vw', height: '100vh' }], { duration: 300 }).finished;
+    }
+
+    // Once the preview is now displayed as the whole viewport, activate.
+    // This performs the instant navigation/URL bar update/etc.
+    try {
+      portal.activate();
+    } catch {
+      // If activation failed, fall back to a normal navigation.
+      location.href = link.href;
```

Comment:
If the reason activation failed is that the user caused a navigation, then this would likely also be ignored. So, it'd be good to return the page to a usable state as well.

For example, consider the following sequence:
User presses back and the back navigation is slow to load, the portal activates and is rejected, the page attempts to navigate and the navigation is ignored, then the user presses stop to stay on the current page. We're left with a portal filling the page.

Reply:
Done, thanks again!